pkg/ring/ring.go

```diff
@@ -195,47 +212,60 @@ func (r *Ring) getInternal(key uint32, n int, op Operation) ([]*IngesterDesc, er
 		distinctHosts[token.Ingester] = struct{}{}
 		ingester := r.ringDesc.Ingesters[token.Ingester]
 
-		// Ingesters that are not ACTIVE do not count to the replication limit. We do
-		// not want to Write to them because they are about to go away, but we do
-		// want to write the extra replica somewhere.  So we increase the size of the
-		// set of replicas for the key.  This means we have to also increase the
+		// We do not want to Write to Ingesters that are not ACTIVE, but we do want
+		// to write the extra replica somewhere.  So we increase the size of the set
+		// of replicas for the key. This means we have to also increase the
 		// size of the replica set for read, but we can read from Leaving ingesters,
 		// so don't skip it in this case.
+		// NB dead ingester will be filtered later (by replication_strategy.go).
```

Comment:
dead and not-`ACTIVE` ingesters are filtered by `IsHealthy()`, which doesn't really match the logic here.  Or I am missing the subtlety.